ApplicationInsights/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -159,35 +160,38 @@ export class TelemetryContext implements ITelemetryContext {
         }
     }
 
-    private _track(envelope: IEnvelope) {
+    private _track(telemetryItem: ITelemetryItem) {
+        let tagsItem: { [key: string]: any } = {};
 
         if (this.session) {
             // If customer set id, apply his context; otherwise apply context generated from cookies 
             if (typeof this.session.id === "string") {
-                this._applySessionContext(envelope, this.session);
+                this._applySessionContext(tagsItem, this.session);
             } else {
-                this._applySessionContext(envelope, this._sessionManager.automaticSession);
+                this._applySessionContext(tagsItem, this._sessionManager.automaticSession);
             }
         }
 
-        this._applyApplicationContext(envelope, this.application);
-        this._applyDeviceContext(envelope, this.device);
-        this._applyInternalContext(envelope, this.internal);
-        this._applyLocationContext(envelope, this.location);
-        this._applySampleContext(envelope, this.sample);
-        this._applyUserContext(envelope, this.user);
-        this._applyOperationContext(envelope, this.operation);
+        // set Part A fields
+        this._applyApplicationContext(tagsItem, this.application);
+        this._applyDeviceContext(tagsItem, this.device);
+        this._applyInternalContext(tagsItem, this.internal);
+        this._applyLocationContext(tagsItem, this.location);
+        this._applySampleContext(tagsItem, this.sample);
+        this._applyUserContext(tagsItem, this.user);
+        this._applyOperationContext(tagsItem, this.operation);
+        telemetryItem.tags.push(tagsItem);
```

Comment:
does this create a nested object inside tags?

Reply:
discussed with Piyali offline. Keeping it as is for now until design is finalized.

---
In reply to: [205518372](https://github.com/Microsoft/ApplicationInsights-JS/pull/635#discussion_r205518372) [](ancestors = 205518372)